ctapipe/reco/HillasReconstructor.py

```diff
@@ -99,13 +99,70 @@ class that reconstructs the direction of an atmospheric shower
 
     """
 
-    def __init__(self, config=None, parent=None, **kwargs):
+    def __init__(self, subarray, config=None, parent=None, **kwargs):
         super().__init__(config=config, parent=parent, **kwargs)
-        self.hillas_planes = {}
-        self.divergent_mode = False
-        self.corrected_angle_dict = {}
+        self.subarray = subarray
+
+        self._cam_radius_m = {
+            cam: cam.geometry.guess_radius()
+            for cam in subarray.camera_types
+        }
+
+        telframe = TelescopeFrame()
+        self._cam_radius_deg = {}
+        for cam, radius_m in self._cam_radius_m.items():
+            cam_name = cam.camera_name
+            cam = next((c for c in subarray.camera_types if c.geometry.camera_name == cam_name))
```

Comment:
Why are you working with camera names here? As I have emphasized before, no part of actual analysis should rely on these names. I don't see why this is necessary.

Reply:
I fail to see the need for this expression. Why can you not just use the `cam` variable from the for loop?

Reply:
Because of the check within the iterator, which was suggested by @kosack because of #1556

Reply:
I don't see how this is connected. What does this solve?

You loop over all `subarray.camera_types` (as these are the keys of `self._cam_radius_m`. You then get the first camera geometry from the same thing that has the same name.

It does not make sense.

Reply:
This code does nothing. It behaves completely the same if you remove the two lines above. What did you want to accomplish?

Reply:
And it is wrong if you have two different camera geometries with the same name (e.g. two UNKNOWN)